Package garden.zbarcam

darosior opened the pull request:
This is to avoid the "garden.zbarcam" is not a package ImportError. Any reason why it has been removed ?

AndreMiras commented:
Thanks for the pull request and head up. It was dropped for two reasons:
1. I didn't realise dropping it would generate such error, in which condition do you have it?
2. it was ruining some of the lazy import approach I had, but probably not an issue anymore (as long as CI is passing)

darosior commented:
I had this error when using the update Zbarcam widget with c-simple (https://github.com/darosior/c-simple). Before now I used an old modified version I took from my forked repo and import worked well, but with the new version `__init__.py` was not present anymore and Python would not recognize it as a package that you can import anymore. I import it like this :
```
from kivy.garden.zbarcam import ZBarCam
```

AndreMiras commented:
OK that makes sense, I usually install it with the other method via pip because it gives more control over the version. So I:
```
pip install --upgrade https://github.com/kivy-garden/garden.zbarcam/archive/develop.zip
```
Then I import with:
```
#:import ZBarCam zbarcam
```
I need to think of a way to integration test both approaches to avoid further regressions

darosior committed:
Make garden.zbarcam a package again

AndreMiras committed:
Update __init__.py adds docstring

- add docstring to explain why this file is required
- add `#noqa` to skip linting

AndreMiras commented:
Build is failing, we need to investigate why to fix and merge. Error is:
```
==================================== ERRORS ====================================
____________________ ERROR collecting tests/test_zbarcam.py ____________________
ImportError while importing test module '/home/user/tests/test_zbarcam.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
tests/test_zbarcam.py:8: in <module>
    from zbarcam import ZBarCam
E   ModuleNotFoundError: No module named 'zbarcam'
------------------------------- Captured stderr --------------------------------
[INFO   ] [Logger      ] Record log in /home/user/.kivy/logs/kivy_19-03-03_1.txt
[INFO   ] [Kivy        ] v1.11.0.dev0, git-Unknown, 20190303
[INFO   ] [Python      ] v3.6.7 (default, Oct 22 2018, 11:32:17) 
[GCC 8.2.0]
[INFO   ] [Factory     ] 184 symbols loaded
[INFO   ] [Image       ] Providers: img_tex, img_dds, img_sdl2, img_pil, img_gif (img_ffpyplayer ignored)
!!!!!!!!!!!!!!!!!!! Interrupted: 1 errors during collection !!!!!!!!!!!!!!!!!!!!
=========================== 1 error in 0.23 seconds ============================
```

AndreMiras commented:
Thanks! Merged after fixing the tox file in fa22533
